Amber-Crystal/amber_cmd
# generate mailers

Here is the procedure to test mailers:

1. create a new project
```
amber new blog --deps
```
2. generate scaffolding for a user
```
amber g scaffold user name:string email:string
```
3. generate a welcome mailer
```
amber g mailer welcome name:string email:string
```
4. send the email when a new user is created by adding the `deliver` to the create method
```
  def create
  user = User.new(params.to_h.select(["name", "email"]))

    if user.valid? && user.save
      WelcomeMailer.new.deliver(name: user.name.not_nil!, email: user.email.not_nil!)
      flash["success"] = "Created User successfully."
      redirect_to "/users"
    else
      flash["danger"] = "Could not create User!"
      render("new.slang")
    end
  end
```
5. launch application using docker-compose
```
dc up -d
```
6. create a new user at `open http://localhost:3000/users/new`

7. check mail catcher that an email was created `open http://localhost:1080`
